lib/rib/extra/autoindent.rb

```diff
@@ -35,7 +35,8 @@ module Rib; module Autoindent
     /^if\b/            => /^(end)\b|^else$|^elsif\b/,
     /^unless\b/        => /^(end)\b|^else$|^elsif\b/,
     /^case\b/          => /^(end)\b|^else$|when\b/  ,
-    /^def\b/           => /^(end)\b/                ,
+    /^def\b/           => 
+      /^(end)\b|^rescue *((\w+)? *(=> *\w+)?)?$/    ,
```

Comment:
Methods can't use `else`, but they could still use `ensure`. Let's also add `ensure` here. To make both regular expressions more consistent, we could change the one for `begin` to:

``` ruby
/^(end)\b|^ensure$|^rescue *((\w+)? *(=> *\w+)?)?$|^else$/
```

And then we could use this here:

``` ruby
/^(end)\b|^ensure$|^rescue *((\w+)? *(=> *\w+)?)?$/
```

What do you think?

Reply:
Actually, giving it a second thought, a method could also use `else` clause, they're actually the same. Let's just make this the same as `begin` then.

Reply:
Thanks!! It is a really nice review and proposal.
I will modify the previous commit to have it added `else` and `ensure` clause also.